indexer/ftypes_matcher.cpp

```diff
@@ -594,50 +593,23 @@ bool IsBridgeOrTunnelChecker::IsMatched(uint32_t type) const
 
 IsHotelChecker::IsHotelChecker()
 {
-  Classificator const & c = classif();
-  for (size_t i = 0; i < static_cast<size_t>(Type::Count); ++i)
-  {
-    auto const hotelType = static_cast<Type>(i);
-    auto const * const tag = GetHotelTypeTag(hotelType);
-    auto const type = c.GetTypeByPath({"tourism", tag});
-
-    m_types.push_back(type);
-
-    m_sortedTypes[i].first = type;
-    m_sortedTypes[i].second = hotelType;
-  }
-
-  sort(m_sortedTypes.begin(), m_sortedTypes.end());
-}
-
-unsigned IsHotelChecker::GetHotelTypesMask(FeatureType & ft) const
-{
-  feature::TypesHolder types(ft);
-  buffer_vector<uint32_t, feature::kMaxTypesCount> sortedTypes(types.begin(), types.end());
-  sort(sortedTypes.begin(), sortedTypes.end());
-
-  unsigned mask = 0;
-  size_t i = 0;
-  size_t j = 0;
-  while (i < sortedTypes.size() && j < m_sortedTypes.size())
-  {
-    if (sortedTypes[i] < m_sortedTypes[j].first)
-    {
-      ++i;
-    }
-    else if (sortedTypes[i] > m_sortedTypes[j].first)
-    {
-      ++j;
-    }
-    else
-    {
-      mask |= 1U << static_cast<unsigned>(m_sortedTypes[j].second);
-      ++i;
-      ++j;
-    }
-  }
+  base::StringIL const types[] = {
+    {"tourism", "alpine_hut"},
+    {"tourism", "apartment"},
+    {"tourism", "camp_site"},
+    {"tourism", "caravan_site"},  /// @todo Sure here?
```

Comment:
Yes, I've stayed in such places. Some of them even have rooms )

> A caravan site, caravan park, [motorhome stopover](https://en.wikipedia.org/wiki/Motorhome_stopover) location or [RV park](https://en.wikipedia.org/wiki/RV_park) is a place where people with caravans / motorhomes / recreational vehicles can stay overnight, or longer, in allotted spaces known as "pitches" or "sites". They usually provide facilities including toilets, waste disposal, water supply, power supply etc..

> They may also have some space for tents.

Reply:
and need to remember to add a basic_hut and a lean-to (both are intended for night overs) when they're ready in https://github.com/organicmaps/organicmaps/pull/7785/files
(depending what's merged first)

Reply:
Shelters are not hotels, definitely. Even if I can sleep inside some shelter.

Reply:
camp_sites are not hotels either ;)
I'm just talking about consistency here

Reply:
Ok, probably criteria should be like: IsHotelChecker returns true for objects that can be booked on Booking or Kayak :)

Reply:
Honestly saying, this is the main purpose of this class.

Reply:
Yeap sorry I've confused it with categories.txt :)
Though some camp sites or caravan sites are bookable (but maybe not on those platforms - IDK).

Reply:
Of course many camp sites and caravan sites are bookable. I've personally booked them in different countries.